File: SCKExample-ObjC/EventEngine.m
//
//  EventEngine.m
//  ScheduleKit
//
//  Created by Guillem Servera Negre on 14/11/16.
//  Copyright © 2016 Guillem Servera. All rights reserved.
//

#import "EventEngine.h"

@import AppKit;

static EventEngine * __shared;

@implementation EventEngine

- (instancetype)init {
    self = [super init];
    if (self) {
        _users = @[
                   [[TestUser alloc] initWithName:@"Dr. Test 1" color:[NSColor systemOrangeColor]],
                   [[TestUser alloc] initWithName:@"Dr. Test 2" color:[NSColor systemBlueColor]]
        ];
        _events = [TestEvent sampleEventsForUsers:_users];
    }
    return self;
}

+ (nonnull instancetype)sharedEngine {
    if (__shared == nil) {
        __shared = [[EventEngine alloc] init];
    }
    return __shared;
}


@end

File: SCKExample-Swift/EventEngine.swift
//
//  EventEngine.swift
//  ScheduleKit
//
//  Created by Guillem Servera Negre on 2/11/16.
//  Copyright © 2016 Guillem Servera. All rights reserved.
//

import Cocoa
import ScheduleKit

extension NSNotification.Name {
    static var eventCountChanged = NSNotification.Name("eventCountChangedNotification")
}

final class EventEngine {

    var events: [TestEvent]
    var users: [TestUser]

    static var shared: EventEngine = EventEngine()
    private init() {
        users = [
            TestUser(name: "Test user 1", color: NSColor.systemOrange),
            TestUser(name: "Test user 2", color: NSColor.systemBlue)
        ]
        events = TestEvent.sampleEvents(for: users)
    }

    func notifyUpdates() {
        NotificationCenter.default.post(name: .eventCountChanged, object: self)
    }
}
